envoy-curses/src/clusters_pane.js

```diff
@@ -0,0 +1,261 @@
+      border: { type: 'line', fg: Theme.style.table.border },
+      columnSpacing: 2,
+      columnWidth: [20, 8, 8, 8, 8, 8],
+    })
+
+    this.connectionsLine = contrib.line({
+      label: 'Stats',
+      showLegend: true,
+      top: 3,
+      left: '50%',
+      width: '50%',
+      height: '100%-3',
+      border: { type: 'line', fg: Theme.style.table.border },
+      legend: { width: 20 },
+      style: Theme.style.chart,
+    })
+
+    const searchStyle = Object.assign(
+      {
+        item: {
+          hover: {
+            bg: Theme.style.base.fg,
+          },
+        },
+        selected: {
+          bg: Theme.style.base.focus.bg,
+          fg: Theme.style.base.focus.fg,
+          bold: true,
+        },
+      },
+      Theme.style.base
+    )
+
+    this.statSearch = blessed.List({
+      label: 'Stats',
+      width: '50%',
+      height: '50%',
+      top: 'center',
+      left: 'center',
+      hidden: true,
+      style: searchStyle,
+      border: { type: 'line', fg: Theme.style.base.border.fg },
+      keys: true,
+      interactive: true,
+    })
+
+    /**
+     * we've selected a new cluster, so update available stats, the underlying
+     * chart model, and update the view
+     */
+    this.clustersTable.rows.on('select', cluster => {
+      this.selectedClusterName = cluster.content.split(/\s+/)[0]
+      this.updateAvailableStats()
+      this.updateChartData()
+      this.updateView()
+    })
+
+    /**
+     * our cluster model is updated, so update the chart and table, and render the view
+     */
+    this.clusters.on('updated', () => {
+      this.updateChartData()
+      this.updateTableData()
+      this.updateView()
+    })
+
+    /**
+     * handle '/' and '?' to launch the stat selection
+     */
+    this.on('element keypress', (ch, key) => {
+      if (!this.detached) {
+        if (key === '/' || key === '?') {
+          this.statSearch.focus()
+          this.statSearch.show()
+          this.screen.render()
+          this.statSearch.once('action', (el, selected) => {
+            this.statSearch.hide()
+            if (el) {
+              this.selectStat(el.content)
+            }
+            this.clustersTable.focus()
+            this.updateChartData()
+            this.updateView()
+          })
+        }
+      }
+    })
+
+    this.on('attach', () => {
+      this.clustersTable.focus()
+    })
+  }
+
+  selectStat(s) {
+    if (s) {
+      this.chartedStat = s
+      this.updateAvailableStats()
+    }
+  }
+
+  /**
+   * build a set of all available stats for the currently selected cluster
+   * @returns {null} nothing
+   */
+  updateAvailableStats() {
+    const hostNames = this.clusters.getHostNames(this.selectedClusterName)
+    const newStats = new Set()
+    for (let i = 0; i < hostNames.length; i++) {
+      if (!this.clusterLevelStats.has(hostNames[i])) {
+        this.clusters.getStatNames(this.selectedClusterName, hostNames[i]).forEach(s => {
+          newStats.add(s)
```

Comment:
i think we can check here if the stat is not numeric and if so then hide it. the current behavior when i select an unrenderable stat (i.e. `health_flags`) is to display an updated title but not an updated the graph.

Reply:
fixed